numba/np/ufunc/dufunc.py

```diff
@@ -496,17 +497,19 @@ def ol_reduce(ufunc, array, axis=0, dtype=None, initial=None):
                 msg = 'The first argument "array" must be array-like'
                 raise errors.NumbaTypeError(msg)
 
-            axis_int = isinstance(axis, types.Integer)
+            axis_tuple = isinstance(axis, types.BaseTuple)
             axis_int_tuple = isinstance(axis, types.UniTuple) and \
                 isinstance(axis.dtype, types.Integer)
             axis_empty_tuple = isinstance(axis, types.Tuple) and len(axis) == 0
             axis_none = cgutils.is_nonelike(axis)
-            axis_tuple_size = len(axis) if axis_int_tuple else 0
 
-            if self.ufunc.identity is None and not (
-                    (axis_int_tuple and axis_tuple_size == 1) or
-                    axis_empty_tuple or axis_int or axis_none):
-                msg = (f"reduction operation '{self.ufunc.__name__}' is not "
+            identity_none = self.ufunc.identity is None
+            ufunc_name = self.ufunc.__name__
+
+            # In NumPy, a ufunc is reorderable if its identity type is **not**
+            # PyUfunc_None.
+            if not self.reorderable and axis_tuple and len(axis) > 1:
```

Comment:
`UniTuple` does not inherit from `BaseTuple` so this might miss catching axes tuples that are of type other than `int` but of a uniform tuple nature ?

Reply:
Fixed!